app/Common/MedusaCommon.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Common;
+
+use Illuminate\Support\Arr;
+
+/**
+ * Trait MedusaCommon.
+ *
+ * Common utility methods
+ */
+trait MedusaCommon
+{
+    /**
+     * Function filterArray.
+     *
+     * Filter an array using a regular expression.  Return only elements of the array that the key matches the regex
+     *
+     * @param array  $array
```

Comment:
Going forward, we should also start saying something about the parameter or return, even for code such as this.

Reply:
@cinnion  You mean a developer might not now that the function `filterArray` that has a param of the type of `array` might be something other than the array you want filtered?  Say it isn't so!

Reply:
Well, more like things that the array is or is not modified, what type of item is in the array (such as required properties for an object), etc.

Reply:
It's a generic function that can be used to filter any array, which is why it's in the MedusaCommon trait, so the only requirement for that parameter is that it be an array.  This will work on both an associative and a numerically indexed array with the right regex.